include/crab/domains/mru_region_domain.hpp

```diff
@@ -0,0 +1,1336 @@
+  using typename abstract_domain_t::variable_or_constant_t;
+  using typename abstract_domain_t::variable_or_constant_vector_t;
+  using typename abstract_domain_t::variable_t;
+  using typename abstract_domain_t::variable_vector_t;
+  using number_t = typename Params::number_t;
+  using varname_t = typename Params::varname_t;
+
+private:
+  using base_uf_domain_t = uf_domain<number_t, varname_t>;
+  using base_abstract_domain_t = typename Params::base_abstract_domain_t;
+  using base_variable_vector_t =
+      typename base_abstract_domain_t::variable_vector_t;
+  using base_bool_domain_t =
+      flat_boolean_numerical_domain<base_abstract_domain_t>;
+  using base_variable_t = typename base_abstract_domain_t::variable_t;
+  using base_variable_or_constant_t =
+      typename base_abstract_domain_t::variable_or_constant_t;
+  using base_linear_expression_t =
+      typename base_abstract_domain_t::linear_expression_t;
+  using base_linear_constraint_t =
+      typename base_abstract_domain_t::linear_constraint_t;
+  using base_linear_constraint_system_t =
+      typename base_abstract_domain_t::linear_constraint_system_t;
+  using base_varname_allocator_t = typename Params::varname_allocator_t;
+  friend class region_domain<Params>;
+  using region_domain_t = region_domain<Params>;
+
+  /**------------------ Begin type definitions ------------------**/
+  using ghost_variables_t = region_domain_impl::ghost_variables<
+      abstract_domain_t, base_abstract_domain_t, base_varname_allocator_t>;
+  // Map from variable to its ghost variables
+  using var_map_t = std::unordered_map<variable_t, ghost_variables_t>;
+  // Reverse map used only for pretty printing: a CrabIR variable has
+  // associated a set of ghost variables. This reverse map maps back
+  // each ghost variable to its CrabIR variable.
+  using ghost_var_id = unsigned;
+  using rev_var_map_t =
+      std::unordered_map<base_variable_t, std::pair<variable_t, ghost_var_id>>;
+  using base_dom_binop_t = std::function<base_abstract_domain_t(
+      base_abstract_domain_t, base_abstract_domain_t)>;
+  using base_uf_dom_binop_t =
+      std::function<base_uf_domain_t(base_uf_domain_t, base_uf_domain_t)>;
+  /**------------------ End type definitions ------------------**/
+
+  /**------------------ Begin field definitions ------------------**/
+  /*
+      abs_domain cache_lines;
+      uf_domain regs;
+      uf_domain addrs;
+      bool_domain flags;
+  */
+  bool m_is_bottom; // special symbol for bottom
+
+  // To create ghost variables for the base domain.
+  base_varname_allocator_t m_alloc;
+  // Map a variable_t to its ghost variables:
+  var_map_t m_var_map;
+  // Reverse map from ghost variables to variable_t
+  rev_var_map_t m_rev_var_map;
+
+public: // FIXME:
+  // The base abstract domains (over ghost variables): the domains only
+  // contain variables
+  base_abstract_domain_t m_cache_lines;
+  base_uf_domain_t m_regs;
```

Comment:
Add comment that says that "it models equalities between scalar variables in `m_mem` and variables in `m_cache_lines`